xcm/xcm-builder/src/barriers.rs

```diff
@@ -39,6 +43,8 @@ impl ShouldExecute for TakeWeightCredit {
 
 /// Allows execution from `origin` if it is contained in `T` (i.e. `T::Contains(origin)`) taking payments into
 /// account.
+///
+/// Only allows for asset-based XCMs.
```

Comment:
this comment is not fully clear to me.

I guess you are trying to say any xcm with `Asset` in the name is matched here, otherwise it is err?

Reply:
it is not clear to me why this is the case, other than some specific implementation. Maybe the name of this struct is too generic